### PR TITLE
corsixth: vendor lua resources

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Like https://github.com/Homebrew/homebrew-core/pull/12326, but actually safe to merge if desired.

### CODE REVIEW

File: Formula/corsixth.rb [hunk 1]
```diff
@@ -1,30 +1,73 @@
...
   depends_on "ffmpeg"
   depends_on "freetype"
   depends_on "lua"
   depends_on "sdl2"
   depends_on "sdl2_mixer"
 
-  depends_on "lpeg" => :lua
-  depends_on "luafilesystem" => :lua
+  resource "lpeg" do
+    url "http://www.inf.puc-rio.br/~roberto/lpeg/lpeg-1.0.1.tar.gz"
```
Comment (ilovezfs) [inline Formula/corsixth.rb]: Looks like there are some https mirrors out there:
https://ftp.openbsd.org/pub/OpenBSD/distfiles/lpeg-1.0.1.tar.gz
https://ftp.heanet.ie/mirrors/ftp.openbsd.org/distfiles/lpeg-1.0.1.tar.gz

File: Formula/corsixth.rb [hunk 2]
```diff
@@ -1,30 +1,73 @@
...
+    ENV["LUA_CPATH"] = "#{luapath}/lib/lua/5.2/?.so"
+
+    resources.each do |r|
+      r.stage do
+        system "luarocks", "build", r.name, "--tree=#{luapath}"
+      end
+    end
+
+    # Ensures it uses the intended Lua (5.2) rather than 5.3 or 5.1 or
+    # attempts to use a combination of both, which can happen.
```
Comment (ilovezfs) [inline Formula/corsixth.rb]: "attempting" I think

File: Formula/corsixth.rb [hunk 3]
```diff
@@ -1,30 +1,73 @@
...
+    inreplace "CMake/FindLua.cmake" do |s|
+      s.gsub! "lua53 lua5.3 lua-5.3 liblua.5.3.dylib", ""
+      s.gsub! "include/lua53 include/lua5.3 include/lua-5.3", "include"
+    end
+
     system "cmake", ".", *std_cmake_args
     system "make"
     prefix.install "CorsixTH/CorsixTH.app"
-    bin.write_exec_script "#{prefix}/CorsixTH.app/Contents/MacOS/CorsixTH"
+    (bin/"CorsixTH").write <<-EOS.undent
```
Comment (ilovezfs) [inline Formula/corsixth.rb]: should we use env_script_all_files or write_env_script for this?

File: Formula/corsixth.rb (hunk 1 above)
Comment (DomT4) [inline Formula/corsixth.rb]: Thanks for finding those. Will change.

File: Formula/corsixth.rb (hunk 2 above)
Comment (DomT4) [inline Formula/corsixth.rb]: Yes, makes more sense 👍.

File: Formula/corsixth.rb (hunk 3 above)
Comment (DomT4) [inline Formula/corsixth.rb]: Seems to be an open question here. There's quite a bit of all the above throughout core code. I do prefer `write_env_script` though.

**Comment by DomT4** [comment]
Tweaks pushed.

**Comment by DomT4** [comment]
Oh Yosemite:
```
        Error: 2 problems in 1 formula
corsixth:
  * Stable resource "lpeg": The URL https://ftp.openbsd.org/pub/OpenBSD/distfiles/lpeg-1.0.1.tar.gz is not reachable
  * HEAD resource "lpeg": The URL https://ftp.openbsd.org/pub/OpenBSD/distfiles/lpeg-1.0.1.tar.gz is not reachable
```

Maybe transient. Or maybe Yosemite's `curl` needs to be considered a dead end sooner rather than later.

**Comment by ilovezfs** [comment]
saw the same thing with libressl https://bot.brew.sh/job/Homebrew/job/homebrew-core/view/change-requests/job/PR-12386/1/testReport/junit/brew-test-bot/yosemite/audit_libressl___online/

**Comment by DomT4** [comment]
I'm pretty sure Yosemite has stopped receiving updates, including security updates, right? Maybe it's time to extend [this](https://github.com/Homebrew/brew/blob/6344989ca7d8fadd6aa102ef2a931e9d0c39ec52/Library/Homebrew/brew.sh#L106) & [this](https://github.com/Homebrew/brew/blob/8ea778f7c7ea790694485030d105a92a9416ef33/Library/Homebrew/extend/os/mac/diagnostic.rb#L244) to Yosemite.

**Comment by ilovezfs** [comment]
:man_shrugging:

**Review by ilovezfs — APPROVED** [review]
:see_no_evil:

**Comment by ilovezfs** [comment]
:hospital:

**Comment by DomT4** [comment]
I'll look into the Yosemite `curl` situation. I'll add it to my growing list of PRs & Issues over at `brew` 😄.

**Comment by ilovezfs** [comment]
>I'll look into proposing changes to the Yosemite curl situation. 

pretty sure this change has deliberately been avoided

**Comment by DomT4** [comment]
I don't think it's an unreasonable change at this point since the situation is only going to get worse, but it's up to y'all.

**Comment by ilovezfs** [comment]
>I don't think it's an unreasonable change at this point since the situation is only going to get worse, but it's up to y'all.

I'd still encourage you to open the PR even if it's not necessarily going to sail through.

**Comment by DomT4** [comment]
My fairly uncontroversial PRs have sat there for 5 days so far untouched. Wasn't sure if that was a hint 😄.

**Comment by ilovezfs** [comment]
Mike's been on a business trip I believe.

**Comment by DomT4** [comment]
No worries. Wasn't sure what the maintainership situation in `brew` was these days, in terms of who stuff runs through; haven't spoken to Mike for a couple months, and `brew` changes rapidly 😸.

**Pull request closed without merging**